controllers/organization_controller_sync.go

```diff
@@ -0,0 +1,123 @@
+package controllers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	orgv1 "github.com/appuio/control-api/apis/organization/v1"
+	controlv1 "github.com/appuio/control-api/apis/v1"
+	"github.com/vshn/appuio-keycloak-adapter/keycloak"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+var orgImportAnnot = "keycloak-adapter.vshn.net/importing"
+
+// Sync lists all Keycloak groups in the realm and creates corresponding Organizations if they do not exist
+func (r *OrganizationReconciler) Sync(ctx context.Context) error {
+	log := log.FromContext(ctx)
+
+	gs, err := r.Keycloak.ListGroups(ctx)
+	if err != nil {
+		log.Error(err, "error listing Keycloak groups")
+		return err
```

Comment:
Don't get errors logged in the higher-level execution anyway? Wouldn't that result in logging the same error twice?
If yes, then I'd propose something like:
```suggestion
		return fmt.Errorf("cannot list Keycloak groups: %w",  err)
```